ios/Subscriptions.swift

```diff
@@ -9,29 +9,50 @@ import Foundation
 
 class Subscriptions: NSObject {
   private var db: SubscriptionsDB!
+  private let lockQueue = dispatch_queue_create("com.mozilla.magnet.subscriptions.access", nil)
```

Comment:
In this file is it necessary to lock _and_ have transactions in the other file?  It looks odd in here that we lock around operations that are supposedly atomic?

If the extra lock was necessary, can we move them into `SubscriptionsDB` (and then use only one lock in there for initialisation and operations)?